pennylane/compiler/qjit_api.py

```diff
@@ -17,43 +17,53 @@
+
+    This decorator enables both just-in-time and ahead-of-time compilation,
+    depending on the compiler package and whether function argument type hints
+    are provided.
 
     .. note::
 
-        This is a wrapper around
-        `catalyst.qjit <https://docs.pennylane.ai/projects/catalyst/en/latest/code/api/catalyst.qjit.html>`__.
+        Currently, only the :doc:`Catalyst <catalyst:index>` hybrid quantum-classical
+        compiler is supported. The Catalyst compiler works with the JAX interface
 
+        For more details, please see the Catalyst documentation and :func:`catalyst:.qjit`
+        docstring.
 
-    This decorator enables both just-in-time and ahead-of-time compilation,
-    depending on the compiler package and whether function argument type hints
-    are provided.
+    .. note::
+
+        Catalyst supports compiling QNodes that use ``lightning.qubit``,
+        ``lightning.kokkos``, ``braket.local.qubit``, and ``braket.aws.qubit``
+        devices. It does not support ``default.qubit``.
+
+        Please see the Catalyst documentation for more details on supported
+        devices, operations, and measurements.
 
     Args:
-        compiler(str): name of the compiler package (default value is ``catalyst``)
-        fn (Callable): the quantum or classical function
+        compiler (str): name of the compiler to use for just-in-time compilation
+        fn (Callable): the quantum or classical function to compile
         autograph (bool): Experimental support for automatically converting Python control
             flow statements to Catalyst-compatible control flow. Currently supports Python ``if``,
             ``elif``, ``else``, and ``for`` statements. Note that this feature requires an
             available TensorFlow installation. Please see the
-            `AutoGraph guide <https://docs.pennylane.ai/projects/catalyst/en/latest/dev/autograph.html>`__
-            for more information.
+            :doc:`AutoGraph guide <catalyst:dev/autograph>` for more information.
         target (str): the compilation target
         keep_intermediate (bool): Whether or not to store the intermediate files throughout the
             compilation. If ``True``, intermediate representations are available via the
             :attr:`~.QJIT.mlir`, :attr:`~.QJIT.jaxpr`, and :attr:`~.QJIT.qir`, representing
             different stages in the optimization process.
         verbosity (bool): If ``True``, the tools and flags used by Catalyst behind the scenes are
             printed out.
-        logfile (Optional[TextIOWrapper]): File object to write verbose messages to (default -
+        logfile (TextIOWrapper): File object to write verbose messages to (default is
             ``sys.stderr``).
-        pipelines (Optional(List[Tuple[str,List[str]]])): A list of pipelines to be executed. The
+        pipelines (List[Tuple[str,List[str]]]): A list of pipelines to be executed. The
```

Comment:
no need to include `Optional` here :) we can be a bit more flexible in docstring types.

Reply:
```suggestion
        pipelines (List[Tuple[str, List[str]]]): A list of pipelines to be executed. The
```